### PR TITLE
task-2: implemented delete method, routing, lifecycle hooks, ngFor

### PR DESCRIPTION
**Pull request by homergit**
task-2: implemented delete method, routing, lifecycle hooks, ngFor. 
In addition: load more, filter, pipes, resset.css

### CODE REVIEW

File: src/app/features/course/courses-section/courses-section.component.ts [hunk 1]
```diff
@@ -1,16 +1,199 @@
...
   selector: 'app-section',
   templateUrl: './courses-section.component.html',
   styleUrls: ['./courses-section.component.scss']
 })
-export class CoursesSectionComponent implements OnInit {
+export class CoursesSectionComponent implements OnChanges,
+                                                OnInit,
+                                                DoCheck,
+                                                AfterContentInit,
+                                                AfterContentChecked,
```
Comment (torin4fan) [inline src/app/features/course/courses-section/courses-section.component.ts]: I think is redundant to implement all these dependencies without practical usage

File: src/app/features/course/courses-section/courses-section.component.ts [hunk 2]
```diff
@@ -1,16 +1,199 @@
...
+                                                AfterViewInit,
+                                                AfterViewChecked,
+                                                OnDestroy {
+  coursesToDisplay: Array<Course> = [];
```
Comment (torin4fan) [inline src/app/features/course/courses-section/courses-section.component.ts]: that's ok, but more popular writing is coursesToDisplay: Course[] = [];

File: src/app/core/pipe/filter.pipe.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+export class FilterPipe implements PipeTransform {
+  transform(data: Array<Course>, term: string): Array<Course> {
+    if (!data) {
+      return [];
+    }
+    if (!term) {
+      return data;
+    }
+
+    return data.filter((item: Course) => JSON.stringify(item).toLowerCase().includes(term.toLowerCase()));
```
Comment (torin4fan) [inline src/app/core/pipe/filter.pipe.ts]: would be great to add 'trim' method for this one

**Review by torin4fan — COMMENTED** [review]
Maybe be better to call breadcrumb in course module because we won't be expect this component in 'login' and 'page-not-found' pages

File: src/app/features/course/courses-section/courses-section.component.ts (hunk 1 above)
Comment (homergit) [reply src/app/features/course/courses-section/courses-section.component.ts]: all the dependencies were implemented due to this point:
image
will be removed.

File: src/app/features/course/courses-section/courses-section.component.ts (hunk 2 above)
Comment (homergit) [reply src/app/features/course/courses-section/courses-section.component.ts]: done, thanks!

File: src/app/core/pipe/filter.pipe.ts (hunk 3 above)
Comment (homergit) [reply src/app/core/pipe/filter.pipe.ts]: done, thanks for your comments.

**Pull request merged**